test/libsolidity/syntaxTests/operators/custom/two_argument_function_bound_to_unary_operator.sol

```diff
@@ -0,0 +1,18 @@
+type Int is int128;
+using {
+    bitnot as ~
+} for Int;
+
+function bitnot(Int, Int) pure returns (Int) {
+    return Int.wrap(13);
+}
+
+contract C {
+    function test() public pure {
+        ~Int.wrap(1);
+    }
+}
+
+// ----
+// TypeError 1147: (32-38): The function "bitnot" needs to have exactly one parameter of type Int to be used for the operator ~.
+// TypeError 4907: (186-198): Unary operator ~ cannot be applied to type Int. No matching user-defined operator found.
```

Comment:
I keep getting confused by this message saying `Unary operator ~ cannot be applied to type Int. No matching user-defined operator found`. It's kinda like it's saying that the operator exists but it does not. I understand why but I always have to think about it first and I think the subtle difference might not be noticed by users at all.

What do you think about modifying the message to say `Built-in unary operator ...`? Would that work or would it mess with other messages where the custom operator does exist?

Reply:
Yeah, the unary case looks good. What about the binary one?